cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/common/ExceptionUtils.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.common;
+
+import io.cdap.cdap.api.exception.ErrorCategory;
+import io.cdap.cdap.api.exception.ErrorCategory.ErrorCategoryEnum;
+import io.cdap.cdap.api.exception.ErrorType;
+import io.cdap.cdap.api.exception.ErrorUtils;
+import io.cdap.cdap.api.exception.ProgramFailureException;
+import io.cdap.cdap.api.exception.WrappedStageException;
+import io.cdap.cdap.etl.api.exception.ErrorDetailsProvider;
+import io.cdap.cdap.etl.api.exception.ErrorPhase;
+import javax.annotation.Nullable;
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ * Utility class for handling exceptions.
+ */
+public class ExceptionUtils {
+
+  /**
+   * Gets the {@link ErrorDetailsProvider} from the given {@link Configuration}.
+   *
+   * @param conf the configuration to get the error details provider from.
+   * @return the error details provider.
+   */
+  @Nullable
+  public static ErrorDetailsProvider getErrorDetailsProvider(Configuration conf) {
+    String errorDetailsProviderClassName =
+      conf.get(ErrorDetailsProvider.EXCEPTION_DETAILS_CLASS_NAME_KEY);
+    if (errorDetailsProviderClassName == null) {
+      return null;
+    }
+    try {
+      return (ErrorDetailsProvider) conf.getClassLoader()
+        .loadClass(errorDetailsProviderClassName)
+        .newInstance();
+    } catch (Exception e) {
+      throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategoryEnum.PLUGIN),
```

Comment:
when would this happen? Wondering if it makes sense to log a warning and return some default provider rather than fail here.

Reply:
ack, done.